java/c/src/test/python/integration_tests.py

```diff
@@ -195,7 +196,18 @@ def test_list_array(self):
             # disabled check_metadata since the list internal field name ("item")
             # is not preserved during round trips (it becomes "$data$").
         ), check_metadata=False)
-        
+
+    def test_empty_list_array(self):
+        with pa.BufferOutputStream() as bos:
+            schema = pa.schema([pa.field("f0", pa.list_(pa.int32()), True)])
+            with ipc.new_stream(bos, schema) as writer:
+                src = pa.RecordBatch.from_arrays([pa.array([[]])], schema=schema)
+                writer.write(src)
+        with pa.input_stream(bos.getvalue()) as ios:
+            with ipc.open_stream(ios) as reader:
+                batch = reader.read_next_batch()
+
+        self.round_trip_record_batch(lambda: batch)
```

Comment:
This may look like an overkill, however I could not came with a better way to create a c-data structures which represents a vector with NULL data buffer, which is what this PR is targeting.

Making just regular batch will create non empty buffers, which is not going to test anything.
Using stream forces the c-data to produce 'optimised' NULL data buffer.

Confirmed, the test fails with upstream and passes with this PR's change.

Reply:
I think you can (ab)use ListArray.from_buffers and pyarrow.py_buffer to manually construct such an array from first principles, though this is probably also OK so long as we assert that the batch it produces has the structure we expect

Reply:
I think this is ok indeed.